File: appLogTic/lib/screens/admin/driver_monitor_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../config/theme.dart';
import '../../models/route.dart';
import '../../providers/driver_monitor_provider.dart';

class DriverMonitorScreen extends StatefulWidget {
  const DriverMonitorScreen({super.key});

  @override
  State<DriverMonitorScreen> createState() => _DriverMonitorScreenState();
}

class _DriverMonitorScreenState extends State<DriverMonitorScreen> {
  DriverWithRoutes? _selectedDriver;

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) {
      context.read<DriverMonitorProvider>().loadDriversWithRoutes();
    });
  }

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<DriverMonitorProvider>();

    return Scaffold(
      appBar: AppBar(
        title: Text(_selectedDriver == null
            ? 'Monitor de Choferes'
            : 'Rutas de ${_selectedDriver!.driver.fullName}'),
        backgroundColor: Theme.of(context).colorScheme.primaryContainer,
        foregroundColor: Theme.of(context).colorScheme.onPrimaryContainer,
        leading: IconButton(
          onPressed: () {
            if (_selectedDriver != null) {
              setState(() => _selectedDriver = null);
            } else {
              Navigator.pop(context);
            }
          },
          icon: const Icon(Icons.arrow_back),
        ),
        actions: [
          IconButton(
            onPressed: () => provider.refreshData(),
            icon: const Icon(Icons.refresh),
          ),
        ],
      ),
      body: _selectedDriver == null
          ? _DriversListView(
              drivers: provider.driversWithRoutes,
              onDriverClick: (driver) => setState(() => _selectedDriver = driver),
            )
          : _DriverRoutesDetailView(driverWithRoutes: _selectedDriver!),
    );
  }
}

class _DriversListView extends StatelessWidget {
  final List<DriverWithRoutes> drivers;
  final ValueChanged<DriverWithRoutes> onDriverClick;

  const _DriversListView({required this.drivers, required this.onDriverClick});

  @override
  Widget build(BuildContext context) {
    return ListView.builder(
      padding: const EdgeInsets.all(16),
      itemCount: drivers.length,
      itemBuilder: (context, index) {
        final driverWithRoutes = drivers[index];
        return Padding(
          padding: const EdgeInsets.only(bottom: 12),
          child: _DriverCard(
            driverWithRoutes: driverWithRoutes,
            onClick: () => onDriverClick(driverWithRoutes),
          ),
        );
      },
    );
  }
}

class _DriverCard extends StatelessWidget {
  final DriverWithRoutes driverWithRoutes;
  final VoidCallback onClick;

  const _DriverCard({required this.driverWithRoutes, required this.onClick});

  @override
  Widget build(BuildContext context) {
    return Card(
      elevation: 4,
      child: InkWell(
        onTap: onClick,
        borderRadius: BorderRadius.circular(12),
        child: Padding(
          padding: const EdgeInsets.all(16),
          child: Column(
            children: [
              Row(
                children: [
                  const Icon(Icons.local_shipping, size: 40, color: AppColors.primary),
                  const SizedBox(width: 12),
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          driverWithRoutes.driver.fullName,
                          style: const TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
                        ),
                        Text(
                          'Rol: ${driverWithRoutes.driver.role}',
                          style: const TextStyle(fontSize: 14, color: AppColors.gray600),
                        ),
                      ],
                    ),
                  ),
                ],
              ),
              const SizedBox(height: 12),
              Column(
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      const Text('Progreso', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500)),
                      Text(
                        '${driverWithRoutes.completionPercentage}%',
                        style: const TextStyle(fontSize: 14, fontWeight: FontWeight.bold, color: AppColors.primary),
                      ),
                    ],
                  ),
                  const SizedBox(height: 4),
                  LinearProgressIndicator(
                    value: driverWithRoutes.completionPercentage / 100,
                    backgroundColor: AppColors.gray200,
                    valueColor: AlwaysStoppedAnimation<Color>(Theme.of(context).colorScheme.primary),
                    minHeight: 8,
                  ),
                ],
              ),
              const SizedBox(height: 12),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceAround,
                children: [
                  _StatItem(label: 'Total', value: '${driverWithRoutes.totalRoutes}', color: Theme.of(context).colorScheme.primary),
                  _StatItem(label: 'En Curso', value: '${driverWithRoutes.inProgressRoutes}', color: Theme.of(context).colorScheme.secondary),
                  _StatItem(label: 'Completadas', value: '${driverWithRoutes.completedRoutes}', color: Theme.of(context).colorScheme.tertiary),
                  _StatItem(label: 'Pendientes', value: '${driverWithRoutes.pendingRoutes}', color: Theme.of(context).colorScheme.error),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

class _StatItem extends StatelessWidget {
  final String label;
  final String value;
  final Color color;

  const _StatItem({required this.label, required this.value, required this.color});

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Text(value, style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold, color: color)),
        Text(label, style: const TextStyle(fontSize: 12, color: AppColors.gray600)),
      ],
    );
  }
}

class _DriverRoutesDetailView extends StatelessWidget {
  final DriverWithRoutes driverWithRoutes;

  const _DriverRoutesDetailView({required this.driverWithRoutes});

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Card(
          margin: const EdgeInsets.all(16),
          color: Theme.of(context).colorScheme.secondaryContainer,
          child: Padding(
            padding: const EdgeInsets.all(16),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceAround,
              children: [
                Column(
                  children: [
                    Text(
                      '${driverWithRoutes.completedRoutes}',
                      style: TextStyle(fontSize: 32, fontWeight: FontWeight.bold, color: Theme.of(context).colorScheme.onSecondaryContainer),
                    ),
                    const Text('Completadas', style: TextStyle(fontSize: 12, color: AppColors.gray700)),
                  ],
                ),
                Container(height: 50, width: 1, color: AppColors.gray400),
                Column(
                  children: [
                    Text(
                      '${driverWithRoutes.pendingRoutes}',
                      style: TextStyle(fontSize: 32, fontWeight: FontWeight.bold, color: Theme.of(context).colorScheme.onSecondaryContainer),
                    ),
                    const Text('Pendientes', style: TextStyle(fontSize: 12, color: AppColors.gray700)),
                  ],
                ),
              ],
            ),
          ),
        ),
        Expanded(
          child: ListView.builder(
            padding: const EdgeInsets.symmetric(horizontal: 16),
            itemCount: driverWithRoutes.routes.length,
            itemBuilder: (context, index) {
              final route = driverWithRoutes.routes[index];
              return Padding(
                padding: const EdgeInsets.only(bottom: 12),
                child: Card(
                  elevation: 2,
                  color: route.status == RouteStatus.completed
                      ? Theme.of(context).colorScheme.tertiaryContainer
                      : route.status == RouteStatus.inProgress
                          ? Theme.of(context).colorScheme.secondaryContainer
                          : null,
                  child: Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Expanded(
                              child: Text(
                                route.clientName,
                                style: const TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
                              ),
                            ),
                            Container(
                              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                              decoration: BoxDecoration(
                                color: route.status == RouteStatus.completed
                                    ? Theme.of(context).colorScheme.tertiary
                                    : route.status == RouteStatus.inProgress
                                        ? Theme.of(context).colorScheme.secondary
                                        : Theme.of(context).colorScheme.primary,
                                borderRadius: BorderRadius.circular(4),
                              ),
                              child: Text(
                                route.status == RouteStatus.completed
                                    ? '✓ Completada'
                                    : route.status == RouteStatus.inProgress
                                        ? '→ En Curso'
                                        : '○ Pendiente',
                                style: const TextStyle(fontSize: 12, color: AppColors.white),
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 8),
                        Text(
                          '${route.address}, ${route.city}',
                          style: const TextStyle(fontSize: 13, color: AppColors.gray600),
                        ),
                        if (route.startTime != null) ...[
                          const SizedBox(height: 8),
                          Text('⏰ Inicio: ${route.startTime}', style: const TextStyle(fontSize: 12, color: AppColors.gray600)),
                        ],
                      ],
                    ),
                  ),
                ),
              );
            },
          ),
        ),
      ],
    );
  }
}

File: app/src/main/java/com/example/logtic/ui/screens/DriverMonitorScreen.kt
package com.example.logtic.ui.screens

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.logtic.data.DriverWithRoutes
import com.example.logtic.data.Route
import com.example.logtic.data.RouteStatus
import com.example.logtic.viewmodel.DriverMonitorViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DriverMonitorScreen(
    driverMonitorViewModel: DriverMonitorViewModel,
    onBack: () -> Unit
) {
    var selectedDriver by remember { mutableStateOf<DriverWithRoutes?>(null) }
    
    Scaffold(
        topBar = {
            TopAppBar(
                title = { 
                    Text(
                        text = if (selectedDriver == null) "Monitor de Choferes" else "Rutas de ${selectedDriver?.driver?.fullName}",
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = {
                        if (selectedDriver != null) {
                            selectedDriver = null
                        } else {
                            onBack()
                        }
                    }) {
                        Icon(
                            imageVector = Icons.Default.ArrowBack,
                            contentDescription = "Volver"
                        )
                    }
                },
                actions = {
                    IconButton(onClick = { driverMonitorViewModel.refreshData() }) {
                        Icon(
                            imageVector = Icons.Default.Refresh,
                            contentDescription = "Actualizar"
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.primaryContainer,
                    titleContentColor = MaterialTheme.colorScheme.onPrimaryContainer
                )
            )
        }
    ) { padding ->
        if (selectedDriver == null) {
            // Vista de lista de choferes
            DriversListView(
                drivers = driverMonitorViewModel.driversWithRoutes.value,
                onDriverClick = { selectedDriver = it },
                modifier = Modifier.padding(padding)
            )
        } else {
            // Vista de detalle de rutas del chofer
            DriverRoutesDetailView(
                driverWithRoutes = selectedDriver!!,
                modifier = Modifier.padding(padding)
            )
        }
    }
}

@Composable
fun DriversListView(
    drivers: List<DriverWithRoutes>,
    onDriverClick: (DriverWithRoutes) -> Unit,
    modifier: Modifier = Modifier
) {
    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        items(drivers) { driverWithRoutes ->
            DriverCard(
                driverWithRoutes = driverWithRoutes,
                onClick = { onDriverClick(driverWithRoutes) }
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DriverCard(
    driverWithRoutes: DriverWithRoutes,
    onClick: () -> Unit
) {
    Card(
        onClick = onClick,
        modifier = Modifier.fillMaxWidth(),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            // Header con nombre del chofer
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.padding(bottom = 12.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.LocalShipping,
                    contentDescription = null,
                    modifier = Modifier.size(40.dp),
                    tint = MaterialTheme.colorScheme.primary
                )
                Spacer(modifier = Modifier.width(12.dp))
                Column {
                    Text(
                        text = driverWithRoutes.driver.fullName,
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold
                    )
                    Text(
                        text = "Rol: ${driverWithRoutes.driver.role}",
                        fontSize = 14.sp,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
            
            // Progress bar
            Column(modifier = Modifier.padding(vertical = 8.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(
                        text = "Progreso",
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Medium
                    )
                    Text(
                        text = "${driverWithRoutes.completionPercentage}%",
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.primary
                    )
                }
                Spacer(modifier = Modifier.height(4.dp))
                LinearProgressIndicator(
                    progress = { driverWithRoutes.completionPercentage / 100f },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(8.dp),
                    color = MaterialTheme.colorScheme.primary,
                    trackColor = MaterialTheme.colorScheme.surfaceVariant,
                )
            }
            
            // Estadísticas
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceAround
            ) {
                StatItem(
                    label = "Total",
                    value = driverWithRoutes.totalRoutes.toString(),
                    color = MaterialTheme.colorScheme.primary
                )
                StatItem(
                    label = "En Curso",
                    value = driverWithRoutes.inProgressRoutes.toString(),
                    color = MaterialTheme.colorScheme.secondary
                )
                StatItem(
                    label = "Completadas",
                    value = driverWithRoutes.completedRoutes.toString(),
                    color = MaterialTheme.colorScheme.tertiary
                )
                StatItem(
                    label = "Pendientes",
                    value = driverWithRoutes.pendingRoutes.toString(),
                    color = MaterialTheme.colorScheme.error
                )
            }
        }
    }
}

@Composable
fun StatItem(
    label: String,
    value: String,
    color: androidx.compose.ui.graphics.Color
) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = value,
            fontSize = 24.sp,
            fontWeight = FontWeight.Bold,
            color = color
        )
        Text(
            text = label,
            fontSize = 12.sp,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
fun DriverRoutesDetailView(
    driverWithRoutes: DriverWithRoutes,
    modifier: Modifier = Modifier
) {
    val context = LocalContext.current
    
    Column(
        modifier = modifier
            .fillMaxSize()
    ) {
        // Resumen
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.secondaryContainer
            )
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                horizontalArrangement = Arrangement.SpaceAround
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text(
                        text = driverWithRoutes.completedRoutes.toString(),
                        fontSize = 32.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSecondaryContainer
                    )
                    Text(
                        text = "Completadas",
                        fontSize = 12.sp,
                        color = MaterialTheme.colorScheme.onSecondaryContainer
                    )
                }
                
                HorizontalDivider(
                    modifier = Modifier
                        .height(50.dp)
                        .width(1.dp)
                )
                
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text(
                        text = driverWithRoutes.pendingRoutes.toString(),
                        fontSize = 32.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSecondaryContainer
                    )
                    Text(
                        text = "Pendientes",
                        fontSize = 12.sp,
                        color = MaterialTheme.colorScheme.onSecondaryContainer
                    )
                }
            }
        }
        
        // Lista de rutas
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            items(driverWithRoutes.routes) { route ->
                AdminRouteCard(
                    route = route,
                    onViewLocation = { lat, lon ->
                        val uri = Uri.parse("geo:$lat,$lon?q=$lat,$lon")
                        val intent = Intent(Intent.ACTION_VIEW, uri)
                        intent.setPackage("com.google.android.apps.maps")
                        if (intent.resolveActivity(context.packageManager) != null) {
                            context.startActivity(intent)
                        }
                    }
                )
            }
        }
    }
}

@Composable
fun AdminRouteCard(
    route: Route,
    onViewLocation: (Double, Double) -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
        colors = CardDefaults.cardColors(
            containerColor = when (route.status) {
                RouteStatus.COMPLETED -> MaterialTheme.colorScheme.tertiaryContainer
                RouteStatus.IN_PROGRESS -> MaterialTheme.colorScheme.secondaryContainer
                else -> MaterialTheme.colorScheme.surface
            }
        )
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = route.clientName,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold
                )
                Surface(
                    shape = MaterialTheme.shapes.small,
                    color = when (route.status) {
                        RouteStatus.COMPLETED -> MaterialTheme.colorScheme.tertiary
                        RouteStatus.IN_PROGRESS -> MaterialTheme.colorScheme.secondary
                        else -> MaterialTheme.colorScheme.primary
                    }
                ) {
                    Text(
                        text = when (route.status) {
                            RouteStatus.COMPLETED -> "✓ Completada"
                            RouteStatus.IN_PROGRESS -> "→ En Curso"
                            else -> "○ Pendiente"
                        },
                        fontSize = 12.sp,
                        color = MaterialTheme.colorScheme.onPrimary,
                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp)
                    )
                }
            }
            
            Spacer(modifier = Modifier.height(8.dp))
            
            Text(
                text = "${route.address}, ${route.city}",
                fontSize = 13.sp,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            
            if (route.startTime != null) {
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = "⏰ Inicio: ${route.startTime}",
                    fontSize = 12.sp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            
            if (route.endTime != null) {
                Text(
                    text = "✓ Fin: ${route.endTime}",
                    fontSize = 12.sp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                
                if (route.endLatitude != null && route.endLongitude != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    OutlinedButton(
                        onClick = { onViewLocation(route.endLatitude, route.endLongitude) },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Icon(
                            imageVector = Icons.Default.LocationOn,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("Ver Ubicación Final", fontSize = 12.sp)
                    }
                }
            }
        }
    }
}
